test/SparqlParserTest.cpp

```diff
@@ -10,6 +10,12 @@
 #include "../src/parser/PropertyPathParser.h"
 #include "../src/parser/SparqlParser.h"
 
+const ParsedQuery pqDummy = []() {
+  ParsedQuery pq;
+  pq._prefixes.emplace_back("xsd", "<http://www.w3.org/2001/XMLSchema#>");
+  return pq;
+}();
+
```

Comment:
The right hand side of  line 13 is a `prvalue` *of the same type as the leftHandSide* and the leftHandSide is being initialized/constructed here, so the copy elision from rhs to lhs is mandatory since C++17 and implemented since forever.

The return value on line 16 is a variable of an object type (no reference etc.) that is defined inside the function/lambda, so a move from line 16 to the rhs of line 13 is mandatory, and copy elision/RVO is allowed since C++11. 
So every reasonable compiler with Optimizations on performs 2 copy elisions, So there is only one `ParsedQuery` object in this whole snippet.

Depending on your settings of standard (older than 17 or not), no-elide-constructors and Optimization level you might see up to three copies, but don't reason about them, because they are always elided.